zookeeper-lock/src/main/java/com/netflix/conductor/zookeeper/ZkLock.java

```diff
@@ -0,0 +1,85 @@
+package com.netflix.conductor.zookeeper;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.netflix.conductor.core.utils.Lock;
+import com.netflix.conductor.zookeeper.config.ZookeeperConfiguration;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.curator.RetryPolicy;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.framework.recipes.locks.InterProcessMutex;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.concurrent.TimeUnit;
+
+public class ZkLock implements Lock {
```

Comment:
ZkLock seems ambiguous. Is this is a common practice in Zookeeper? If not, it's probably better to use full names.

Reply:
It is standard shorthand, can spell it out too.  For now will merge first to setup testing.